src/pages/Dashboard/Billing/PaymentForm/index.tsx

```diff
@@ -0,0 +1,169 @@
+/* eslint-disable multiline-ternary */
+// @ts-nocheck
+import React, { useState } from 'react';
+import Cards from 'react-credit-cards-2';
+import TextField from '@mui/material/TextField';
+import InputMask from 'react-input-mask';
+import Button from '../../../../components/Button';
+import { useLocation } from 'react-router-dom';
+
+type Focused = 'name' | 'number' | 'expiry' | 'cvc';
+
+const PaymentForm: React.FC = (): JSX.Element => {
+  const [creditNumber, setCreditNumber] = useState('');
+  const [creditName, setCreditName] = useState('');
+  const [cvc, setCvc] = useState('');
+  const [expiry, setExpiry] = useState('');
+  const [focus, setFocus] = useState<Focused | undefined>();
+  const [saveCreditCardInfo, setSaveCreditCardInfo] = useState(false); // new state for saving credit card info
+  const location = useLocation();
+  const packageInfo = {
+    packageName:
+      (location?.search.includes('package=')
+        ? new URLSearchParams(location.search).get('package')
+        : '') ?? '',
+    cost:
+      (location?.search.includes('cost=')
+        ? new URLSearchParams(location.search).get('cost')
+        : '') ?? '',
```

Comment:
This could be done simply using the navigation library that the application is using `react-router-dom`